app/services/charts/annotate.rb

```diff
@@ -1,52 +1,101 @@
+      weekly_relevant_start_date = date_categories.find { |date| (date..(date + 6.days)).cover?(observation.at.to_date) }
 
-      first_date = date_categories.min
-      last_date = date_categories.max + 6.days
+      x_axis_categories[date_categories.index(weekly_relevant_start_date)]
+    end
 
-      relevant_interventions = @interventions_scope.where('at BETWEEN ? AND ?', first_date, last_date)
+    def daily_relevant_observations_for(x_axis_start, x_axis_end)
+      relevant_observations = relevant_observations_for(Date.parse(x_axis_start), Date.parse(x_axis_end))
 
-      relevant_interventions.map do |intervention|
-        relevant_start_date = date_categories.find {|date| (date..(date + 6.days)).cover?(intervention.at.to_date)}
-        {
-          id: intervention.id,
-          event: intervention.intervention_type.name,
-          date: intervention.at.to_date,
-          x_axis_category: x_axis_categories[date_categories.index(relevant_start_date)],
-          icon: intervention.intervention_type.intervention_type_group.icon
-        }
+      relevant_observations.map do |observation|
+        annotation_for(observation: observation, x_axis_category: observation.at.strftime('%d-%m-%Y'))
       end
     end
 
-    def annotate_daily(x_axis_start, x_axis_end)
-      return if x_axis_start.blank? || x_axis_end.blank?
-      first_date = Date.parse(x_axis_start)
-      last_date = Date.parse(x_axis_end)
-
-      relevant_interventions = @interventions_scope.where('at BETWEEN ? AND ?', first_date, last_date)
-
-      relevant_interventions.map do |intervention|
-        {
-          id: intervention.id,
-          event: intervention.intervention_type.name,
-          date: intervention.at.to_date,
-          x_axis_category: intervention.at.strftime('%d-%m-%Y'),
-          icon: intervention.intervention_type.intervention_type_group.icon
-        }
+    def annotation_for(observation:, x_axis_category:)
+      {
+        id: observation.id,
+        event: event_for(observation),
+        date: observation.at.to_date,
+        x_axis_category: x_axis_category,
+        icon: icon_for(observation),
+        icon_color: icon_color_for(observation),
+        observation_type: observation.observation_type,
+        url: url_for(observation)
+      }
+    end
+
+    def url_for(observation)
+      case observation.observation_type
+      when 'activity' then Rails.application.routes.url_helpers.school_activity_path(@school, observation.activity)
+      when 'intervention' then Rails.application.routes.url_helpers.school_intervention_path(@school, observation)
       end
     end
 
-    private
+    def icon_for(observation)
+      case observation.observation_type
+      when 'activity' then observation.activity.activity_category.icon
+      when 'intervention' then observation.intervention_type.intervention_type_group.icon
+      end
+    end
+
+    def icon_color_for(observation)
```

Comment:
Not sure about the colours. I don't think introducing new colours is best option here and might be distracting.

Can I suggest using $dark and $grey from our standard colours currently, then we can get feedback?